File: HelloWorld/app/src/main/java/com/cardlay/helloworld/DisplayMessageActivity.java
package com.cardlay.helloworld;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.widget.TextView;

public class DisplayMessageActivity extends AppCompatActivity {

    public static final String TAG = "activities_events";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_display_message);


        // Get the Intent that started this activity and extract the string
        Intent intent = getIntent();
        String message = intent.getStringExtra(MainActivity.EXTRA_MESSAGE);

        // Capture the layout's TextView and set the string as its text
        TextView textView = findViewById(R.id.textView);
        textView.setText(message);

        Log.i(TAG, "DisplayMessageActivity onCreate");
    }

    @Override
    protected void onStart() {
        super.onStart();
        Log.i(TAG, "DisplayMessageActivity onStart");
    }

    @Override
    protected void onResume() {
        super.onResume();
        Log.i(TAG, "DisplayMessageActivity onResume");
    }

    @Override
    protected void onPause() {
        super.onPause();
        Log.i(TAG, "DisplayMessageActivity onPause");
    }

    @Override
    protected void onStop() {
        super.onStop();
        Log.i(TAG, "DisplayMessageActivity onStop");
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        Log.i(TAG, "DisplayMessageActivity onDestroy");
    }

    @Override
    protected void onRestart() {
        super.onRestart();
        Log.i(TAG, "DisplayMessageActivity onRestart");
    }
}

File: HelloWorld/app/src/main/java/com/cardlay/helloworld/DisplayMessageActivity.kt
package com.cardlay.helloworld

import android.os.Bundle
import android.util.Log
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class DisplayMessageActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_display_message)

        // Get the Intent that started this activity and extract the string
        val message = intent.getStringExtra(MainActivity.Companion.EXTRA_MESSAGE)

        // Capture the layout's TextView and set the string as its text
        val textView = findViewById<TextView>(R.id.textView)
        textView.text = message
        Log.i(TAG, "DisplayMessageActivity onCreate")
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        Log.i(TAG, "DisplayMessageActivity onSaveInstanceState")
    }

    override fun onRestoreInstanceState(savedInstanceState: Bundle) {
        super.onRestoreInstanceState(savedInstanceState)
        Log.i(TAG, "DisplayMessageActivity onRestoreInstanceState")
    }

    override fun onStart() {
        super.onStart()
        Log.i(TAG, "DisplayMessageActivity onStart")
    }

    override fun onResume() {
        super.onResume()
        Log.i(TAG, "DisplayMessageActivity onResume")
    }

    override fun onPause() {
        super.onPause()
        Log.i(TAG, "DisplayMessageActivity onPause")
    }

    override fun onStop() {
        super.onStop()
        Log.i(TAG, "DisplayMessageActivity onStop")
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.i(TAG, "DisplayMessageActivity onDestroy")
    }

    override fun onRestart() {
        super.onRestart()
        Log.i(TAG, "DisplayMessageActivity onRestart")
    }

    companion object {
        const val TAG = "activities_events"
    }
}